library/kotlin/test/io/envoyproxy/envoymobile/RequestBuilderTest.kt

```diff
@@ -0,0 +1,128 @@
+package io.envoyproxy.envoymobile
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import java.net.URL
+
+class RequestBuilderTest {
+
+  @Test
+  fun `adding request data should have body present in request`() {
+    val body = "data".toByteArray()
+
+    val request = RequestBuilder(URL("http://0.0.0.0:9001/api.lyft.com/demo.txt"), RequestMethod.GET)
+        .addBody(body)
+        .build()
+    assertThat(request.body).isEqualTo(body)
+  }
+
+  @Test
+  fun `not adding request data should have null body in request`() {
+    val request = RequestBuilder(URL("http://0.0.0.0:9001/api.lyft.com/demo.txt"), RequestMethod.GET)
+        .build()
+
+    assertThat(request.body).isNull()
+  }
+
+  @Test
+  fun `adding retry policy should have policy present in request`() {
+
+    val retryPolicy = RetryPolicy(23, listOf(RetryRule.FIVE_XX, RetryRule.CONNECT_FAILURE), 1234)
+    val request = RequestBuilder(URL("http://0.0.0.0:9001/api.lyft.com/demo.txt"), RequestMethod.GET)
+        .addRetryPolicy(retryPolicy)
+        .build()
+
+    assertThat(request.retryPolicy).isEqualTo(RetryPolicy(23, listOf(RetryRule.FIVE_XX, RetryRule.CONNECT_FAILURE), 1234))
```

Comment:
Why not compare directly to `retryPolicy`?

Reply:
It's easier to see the values which we are expecting on an assert line